src/core_modules/capture-core/components/Pages/MainPage/WorkingLists/ListViewLoader/ListViewLoader.component.js

```diff
@@ -1,22 +1,114 @@
 // @flow
-import React, { memo } from 'react';
+import React, { memo, useCallback, useContext, useEffect, useRef } from 'react';
 import { withLoadingIndicator, withErrorMessageHandler } from '../../../../../HOC';
 import { ListViewUpdater } from '../ListViewUpdater';
 import { ListViewLoaderContext } from '../workingLists.context';
+import { useIsContextInSync } from './useIsContextInSync';
 import type { Props } from './listViewLoader.types';
 
 const EventListUpdaterWithLoadingIndicator = withErrorMessageHandler()(
     withLoadingIndicator(() => ({ margin: 10 }))(ListViewUpdater));
 
-// eslint-disable-next-line complexity
+const hasTemplateChange = (currentTemplate, prevTemplate, viewPreloaded) =>
+    (prevTemplate && currentTemplate.id !== prevTemplate.id && !viewPreloaded);
+
+const useCalculateTriggerLoad = ({
```

Comment:
Perhaps write a comment above explaining which cases `useCalculateTriggerLoad`. It is a little bit difficult to follow for me at least